compiler-plugin/src/main/kotlin/arrow/meta/quotes/ClassOrObject.kt

```diff
@@ -13,12 +13,112 @@ import org.jetbrains.kotlin.psi.psiUtil.getValueParameters
 import org.jetbrains.kotlin.psi.psiUtil.modalityModifierType
 import org.jetbrains.kotlin.psi.psiUtil.visibilityModifierType
 
+/**
+ * [classOrObject] is a function that intercepts all [KtClass] elements that [match]
+ * allowing a [Transform] to change the intercepted AST tree before compilation.
```

Comment:
```suggestion
* then uses a [Transform] to change the intercepted AST tree before compilation.
```
💡 nitpick, but it seems easier to understand for me like this, when you're not aware of what's a `Transform` in the first place.

Reply:
Thanks Jorge! Much appreciated.